packages/bcer-api/app/src/notification/subscription.controller.ts

```diff
@@ -0,0 +1,68 @@
+import {
+  Body,
+  Controller,
+  ForbiddenException,
+  Get,
+  HttpCode,
+  HttpStatus,
+  Patch,
+  Request,
+  UseGuards,
+} from '@nestjs/common';
+import {
+  ApiBearerAuth,
+  ApiOperation,
+  ApiResponse,
+  ApiTags,
+} from '@nestjs/swagger';
+import { AuthGuard, RoleGuard, Roles } from 'src/auth/auth.module';
+import { RequestWithUser } from 'src/auth/interface/requestWithUser.interface';
+import { SubscriptionDTO } from './dto/subscription.dto';
+import { NotificationService } from './notification.service';
+import { SubscriptionRO } from './ro/subscription.ro';
+
+@ApiBearerAuth()
+@UseGuards(AuthGuard, RoleGuard)
+@ApiTags('subscription')
+@Controller('subscription')
+export class SubscriptionController {
+  constructor(private notificationService: NotificationService) {}
+
+  @ApiOperation({
+    summary: 'Creates or updates a subscription to receive notification',
+  })
+  @ApiResponse({ status: HttpStatus.CREATED })
+  @ApiBearerAuth()
+  @HttpCode(HttpStatus.CREATED)
+  @Roles('user')
+  @Patch()
+  async subscribeToTextNotification(
+    @Request() req: RequestWithUser,
+    @Body() payload: SubscriptionDTO,
+  ): Promise<SubscriptionRO> {
+    if (req.ctx.businessId !== payload.businessId) {
+      throw new ForbiddenException(null, 'user does not have permission to update this business');
+    }
+    const subscription = await this.notificationService.createOrUpdateSubscription(
+      payload,
+    );
+    return subscription.toResponseObject();
+  }
+
+  @ApiOperation({
+    summary: 'gets the subscription for a business if any',
+  })
+  @ApiResponse({ status: HttpStatus.OK })
+  @ApiBearerAuth()
+  @HttpCode(HttpStatus.OK)
+  @Roles('user')
+  @Get()
+  async getSubscription(
+    @Request() req: RequestWithUser,
+  ): Promise<SubscriptionRO> {
+    const subscription = await this.notificationService.findSubscription(
+      req.ctx.businessId,
+    );
```

Comment:
Think we might need the `if (req.ctx.businessId !== payload.businessId) {` check here as well to prevent you from looking up the subscription details for any other businesses

Reply:
Actually, there is nothing in the payload, instead, we are using the `req.ctx.businessId` directly so we should be ok?

Reply:
Ah, you're right. Can discard this